### ISSUE TITLE
matrix module eigenvector method simplify flag will not apply to multiplicities gt 1

### ISSUE THREAD

**Issue by AncientZygote** [sympy/sympy#13124]
_sympy.matrices_ invokes the _eigenvects_ method with default _simplify_=_False_, which leaves possible fractions, i.e., pure integers in the denominators of the eigenvector values.

For a matrix **A**, _A.eigenvects_ (_simplify_=_True_) is supposed to multiply out any fractional results. This works properly except when the multiplicity of the eigenvector is greater than one. Example:
<pre><code>
AAA = Matrix ( [ [204,98,-26,-10],[-280,-134,36,14],[716,348,-90,-36],[-472,-232,60,28] ] )
</code></pre>
Inspect this input result with srepr:
<pre><code>
srepr(AAA)
MutableDenseMatrix([[Integer(204), Integer(98), Integer(-26), Integer(-10)], [Integer(-280), Integer(-134), Integer(36), Integer(14)], [Integer(716), Integer(348), Integer(-90), Integer(-36)], [Integer(-472), Integer(-232), Integer(60), Integer(28)]])
</code></pre>
<pre><code>
>>> AAA.eigenvects()

⎡⎛0, 1, ⎡⎡-3/4⎤⎤⎞, ⎛2, 2, ⎡⎡1/4 ⎤, ⎡-3/8⎤⎤⎞, ⎛4, 1, ⎡⎡1/5 ⎤⎤⎞⎤
⎢⎜      ⎢⎢    ⎥⎥⎟  ⎜      ⎢⎢    ⎥  ⎢    ⎥⎥⎟  ⎜      ⎢⎢    ⎥⎥⎟⎥
⎢⎜      ⎢⎢ 1  ⎥⎥⎟  ⎜      ⎢⎢-1/4⎥  ⎢7/8 ⎥⎥⎟  ⎜      ⎢⎢-1/5⎥⎥⎟⎥
⎢⎜      ⎢⎢    ⎥⎥⎟  ⎜      ⎢⎢    ⎥  ⎢    ⎥⎥⎟  ⎜      ⎢⎢    ⎥⎥⎟⎥
⎢⎜      ⎢⎢-5/2⎥⎥⎟  ⎜      ⎢⎢ 1  ⎥  ⎢ 0  ⎥⎥⎟  ⎜      ⎢⎢2/5 ⎥⎥⎟⎥
⎢⎜      ⎢⎢    ⎥⎥⎟  ⎜      ⎢⎢    ⎥  ⎢    ⎥⎥⎟  ⎜      ⎢⎢    ⎥⎥⎟⎥
⎣⎝      ⎣⎣ 1  ⎦⎦⎠  ⎝      ⎣⎣ 0  ⎦  ⎣ 1  ⎦⎦⎠  ⎝      ⎣⎣ 1  ⎦⎦⎠⎦
</code></pre>

<pre><code>
>>> AAA.eigenvects(simplify=True)

⎡⎛0, 1, ⎡⎡-3 ⎤⎤⎞, ⎛2, 2, ⎡⎡1 ⎤, ⎡-3/8⎤⎤⎞, ⎛4, 1, ⎡⎡1 ⎤⎤⎞⎤
⎢⎜      ⎢⎢   ⎥⎥⎟  ⎜      ⎢⎢  ⎥  ⎢    ⎥⎥⎟  ⎜      ⎢⎢  ⎥⎥⎟⎥
⎢⎜      ⎢⎢ 4 ⎥⎥⎟  ⎜      ⎢⎢-1⎥  ⎢7/8 ⎥⎥⎟  ⎜      ⎢⎢-1⎥⎥⎟⎥
⎢⎜      ⎢⎢   ⎥⎥⎟  ⎜      ⎢⎢  ⎥  ⎢    ⎥⎥⎟  ⎜      ⎢⎢  ⎥⎥⎟⎥
⎢⎜      ⎢⎢-10⎥⎥⎟  ⎜      ⎢⎢4 ⎥  ⎢ 0  ⎥⎥⎟  ⎜      ⎢⎢2 ⎥⎥⎟⎥
⎢⎜      ⎢⎢   ⎥⎥⎟  ⎜      ⎢⎢  ⎥  ⎢    ⎥⎥⎟  ⎜      ⎢⎢  ⎥⎥⎟⎥
⎣⎝      ⎣⎣ 4 ⎦⎦⎠  ⎝      ⎣⎣0 ⎦  ⎣ 1  ⎦⎦⎠  ⎝      ⎣⎣5 ⎦⎦⎠⎦
>>> AAA.eigenvals()
{0: 1, 2: 2, 4: 1}
</code></pre>

The diagonalize method apparently suffers from the same problem:
<pre><code>
>>> QQ, LL = AAA.diagonalize()
>>> QQ
⎡-3   1   -3/8  1 ⎤
⎢                 ⎥
⎢ 4   -1  7/8   -1⎥
⎢                 ⎥
⎢-10  4    0    2 ⎥
⎢                 ⎥
⎣ 4   0    1    5 ⎦
</code></pre>
Because any nonzero scalar multiple of an eigenvector is again an eigenvector, this behavior is not really an error, however, users should be made aware of it so as not to glance at a result from a sample textbook problem and immediately conclude they are obtaining erroneous results from Sympy. There may be other implications in terms of usage of the result in other context that I am not aware of at the moment. I assume it is merely a matter of modifying the _matrices.py_ code in this loop area:

<pre><code>
if primitive:
                # the relationship A*e = lambda*e will still hold if we change the
                # eigenvector; so if simplify is True we tidy up any normalization
                # artifacts with as_content_primtive (default) and remove any pure Integer
                # denominators.
                l = 1
                for i, b in enumerate(basis[0]):
                    c, p = signsimp(b).as_content_primitive()
                    if c is not S.One:
                        b = c*p
                        l = ilcm(l, c.q)
                    basis[0][i] = b
                if l != 1:
                    basis[0] *= l
</code></pre>

**Comment by cbm755** [comment]
I agree this would be nice to have.  Are interesting in trying the fix?  Looks like you've found a reasonable place to start!

**Comment by AncientZygote** [comment]
I'm a little old and slow, Colin, but I could take a look at a fix in the background (of my daily Bataan Death March, grin). 

Dalton Bentley
<p><a href="https://physics.stackexchange.com/users/111801/dalton-bentley/"> SE profile as bio</a> inline link.</p>
